Lib/unittest/test/testmock/testpatch.py

```diff
@@ -105,6 +105,8 @@ def test():
         self.assertEqual(Something.attribute, sentinel.Original,
                          "patch not restored")
 
+    def test_patchobject_with_string_as_target(self):
+        self.assertRaises(TypeError, patch.object, 'Something', 'do_something')
```

Comment:
Aah actually, Mario's comment about the message reminds me that we should probably check the message here. Could you use something like 

https://github.com/python/cpython/blob/fd5c414880b2e05720b9cf14ab0b0d7ae2b7d925/Lib/test/test_named_expressions.py#L17-L18

to make sure we get the expected message for the TypeError?